File: programme/fibonacci-pattern/fibonacci_pattern.js
/* Write a programme to print the Fibonacci pattern

Example:
Input: n = 5
Output:
1
1 1
1 1 2
1 1 2 3
1 1 2 3 5
*/

const printFibs = (n) => {
  let str = "";
  let a = 1;
  let b = 1;
  for (let i = 0; i < n; i++) {
    str += `${a} `;
    console.log(str);
    [a, b] = [b, a + b];
  }
};

printFibs(5);

File: programme/fibonacci-pattern/fibonacci_pattern.go
/* Write a programmeto print Fibonacci pattern using recursion

Example:
Input: n = 5
Output:
1
1 1
1 1 2
1 1 2 3
1 1 2 3 5
*/

package main

import (
	"fmt"
	"strconv"
)

func fibonacci(n int) int {
	if n <= 1 {
		return n
	}
	return fibonacci(n-1) + fibonacci(n-2)
}

func main() {
	var n int
	fmt.Print("Enter an integer : ")
	fmt.Scan(&n)

	for i := 1; i <= n; i++ {
		for j := 1; j <= i; j++ {
			fmt.Print(strconv.Itoa(fibonacci(j)) + " ")
		}
		fmt.Println()
	}
}
